x-pack/plugin/transform/src/main/java/org/elasticsearch/xpack/transform/transforms/TransformFailureHandler.java

```diff
@@ -51,28 +52,32 @@ class TransformFailureHandler {
     /**
      * Handle a search or indexing failure
      *
-     * @param e the exception caught
+     * @param exception the exception caught
      * @param settingsConfig The settings
      */
-    void handleIndexerFailure(Exception e, SettingsConfig settingsConfig) {
+    void handleIndexerFailure(Exception exception, SettingsConfig settingsConfig) {
         // more detailed reporting in the handlers and below
-        logger.debug(() -> "[" + transformId + "] transform encountered an exception: ", e);
-        Throwable unwrappedException = ExceptionsHelper.findSearchExceptionRootCause(e);
+        logger.atDebug().withThrowable(exception).log("[{}] transform encountered an exception", transformId);
```

Comment:
That's a nice finding :)
I've always struggled with message formatting when I passed the exception to `debug()`